sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -594,16 +594,13 @@ qualifiedName
     : identifier ('.' identifier)*
     ;
 
-// Identifier that also allows the use of a number of SQL keywords (mainly for backwards compatibility).
-looseIdentifier
-    : identifier
-    | FROM
-    | TO
-    | TABLE
-    | WITH
+identifier
+    : strictIdentifier
+    | ANTI | FULL | INNER | LEFT | SEMI | RIGHT | NATURAL | UNION | INTERSECT | EXCEPT | DATABASE | ON | JOIN | CROSS
+    | SELECT | FROM | WHERE | HAVING | FROM | TO | TABLE | WITH | NOT
```

Comment:
I don't have a reference but let me explain. The general issue is if we would make them non-reserved they would introduce ambiguities during the parsing (if you are lucky you get a parsing error or if you are unlucky very weird behavior). 

I can give you an example for joins: `select * from a left join b on b.id = a.id` would be interpreted as *a* **aliased as left** *inner join b using condition b.id = a.id*.

We might not need the other keywords here. Let me check if I can move them.